src/main/java/net/openhft/chronicle/threads/MonitorEventLoop.java

```diff
@@ -130,7 +124,6 @@ public void run() {
     @HotMethod
     private boolean runHandlers() {
         boolean busy = false;
-        // assumed to be synchronized in run()
         for (int i = 0; i < handlers.size(); i++) {
```

Comment:
should we do handlers.forEach here?

Reply:
Would that generate garbage for the lambda?

Reply:
oh, woops, I didn't realise it was in fast path - was confused by the diff. Should not matter if the collection is changed underneath this loop anyway, I believe the worst can happen is a handler is executed twice